dotCMS/src/main/java/com/dotcms/browser/BrowserAPIImpl.java

```diff
@@ -0,0 +1,524 @@
+package com.dotcms.browser;
+
+import com.dotcms.business.CloseDBIfOpened;
+import com.dotcms.contenttype.exception.NotFoundInDbException;
+import com.dotcms.contenttype.model.type.BaseContentType;
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.beans.IconType;
+import com.dotmarketing.beans.Identifier;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.DotStateException;
+import com.dotmarketing.business.PermissionAPI;
+import com.dotmarketing.business.Role;
+import com.dotmarketing.business.web.UserWebAPI;
+import com.dotmarketing.business.web.WebAPILocator;
+import com.dotmarketing.comparators.WebAssetMapComparator;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.contentlet.transform.ContentletToMapTransformer;
+import com.dotmarketing.portlets.fileassets.business.FileAsset;
+import com.dotmarketing.portlets.fileassets.business.FileAssetAPI;
+import com.dotmarketing.portlets.folders.business.FolderAPI;
+import com.dotmarketing.portlets.folders.model.Folder;
+import com.dotmarketing.portlets.htmlpageasset.model.HTMLPageAsset;
+import com.dotmarketing.portlets.languagesmanager.business.LanguageAPI;
+import com.dotmarketing.portlets.languagesmanager.model.Language;
+import com.dotmarketing.portlets.links.model.Link;
+import com.dotmarketing.portlets.workflows.business.WorkflowAPI;
+import com.dotmarketing.portlets.workflows.model.WorkflowAction;
+import com.dotmarketing.portlets.workflows.model.WorkflowScheme;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.UtilHTML;
+import com.dotmarketing.util.UtilMethods;
+import com.liferay.portal.language.LanguageUtil;
+import com.liferay.portal.model.User;
+import com.liferay.util.StringPool;
+import io.vavr.control.Try;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static com.dotmarketing.business.PermissionAPI.PERMISSION_READ;
+
+/**
+ * Default implementation
+ * @author jsanca
+ */
+public class BrowserAPIImpl implements BrowserAPI {
+
+    private final LanguageAPI langAPI       = APILocator.getLanguageAPI();
+    private final UserWebAPI userAPI       = WebAPILocator.getUserWebAPI();
+    private final FolderAPI folderAPI     = APILocator.getFolderAPI();
+    private final PermissionAPI permissionAPI = APILocator.getPermissionAPI();
+
+    @Override
+    @CloseDBIfOpened
+    public Map<String, Object> getFolderContent(final BrowserQuery browserQuery) throws DotSecurityException, DotDataException {
+
+        final boolean respectFrontEndRoles = false;
+        List<Map<String, Object>> returnList = new ArrayList<>();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5337824)